integration/spark/src/main/scala/org/apache/spark/sql/catalyst/CarbonParserUtil.scala

```diff
@@ -337,7 +331,7 @@ object CarbonParserUtil {
 
     if (tableProperties.get(CarbonCommonConstants.COLUMN_META_CACHE).isDefined) {
       // validate the column_meta_cache option
-      val tableColumns = dims.view.filterNot(_.spatialIndex).map(x => x.name.get) ++
+      val tableColumns = dims.view.map(x => x.name.get) ++
```

Comment:
if we had not removed this filter, then you wouldn't require modification in `validateColumnMetaCacheFields` or `validateColumnMetaCacheOption` i guess.

Reply:
yes, made changes.